File: frontend/src/types/address.ts
export interface Address {
    id?: number;
    wardCode: number;
    wardName: string;
    cityCode: number;
    cityName: string;
    detail: string;
    longitude: number;
    latitude: number;
    isDefault: boolean;
    name: string;
    phoneNumber: string;
    fullAddress: string;
    isSaved: boolean;
    type: AddressType;
}

export interface AddressRequest {
    id?: number;
    wardCode: number;
    wardName: string;
    cityCode: number;
    cityName: string;
    detail: string;
    name: string;
    phoneNumber: string;
    isDefault: boolean;
    latitude: number;
    longitude: number;
}

export type AddressType = 'SENDER' | 'RECIPIENT';

File: backend/src/main/java/com/logistics/entity/Address.java
package com.logistics.entity;

import com.logistics.enums.AddressType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.envers.Audited;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Builder
@Entity
@Audited
@Data
@Table(name = "addresses")
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, unique = true)
    private Integer id;

    @Column(nullable = false)
    private Integer wardCode;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String wardName;

    @Column(nullable = false)
    private Integer cityCode;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String cityName;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String detail;

    @Column(columnDefinition = "NVARCHAR(50)")
    private String name;

    @Column(columnDefinition = "NVARCHAR(11)")
    private String phoneNumber;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    private Boolean isDefault;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private AddressType type = AddressType.RECIPIENT;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String fullAddress;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}